packages/ui/src/components/BarChart/BarChart.tsx

```
@@ -0,0 +1,165 @@
+  const chartData = data[0].x.map((date, index) => {
+    const item: Record<string, any> = { date };
+    data.forEach((series) => {
+      item[series.name || "value"] = series.y[index];
+    });
+    return item;
+  });
+
+  // Create chart config with default colors
+  const chartConfig = data.reduce<ChartConfig>(
+    (acc, series, index) => ({
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)